integration-tests/modules/__tests__/product/store/index.spec.ts

```diff
@@ -247,6 +247,18 @@ medusaIntegrationTestRunner({
             ])
           })
 
+          it("should list products by id", async () => {
```

Comment:
actually, now that I think about it, the union extension you added might not work.

If you have `/store/products?id[]=product-not-in-sales-channel&sales_channel_id[]=sc-id`, this should return an empty products response. I'm not sure that will be the case with a union

Reply:
@riqwan what I implemented is an intersection, not a union, so it should behave exactly as you said in the example, the ID has to be both in the id filters and in the sales channels.

Reply:
@riqwan merging as from what I can tell it works as expected, let me know if you found some other issue with the current approach.